src/psyclone/psyGen.py

```diff
@@ -2641,8 +2650,8 @@ def reduction_sum_loop(self, parent):
             api_strings = [access.api_specific_name()
                            for access in REDUCTION_OPERATOR_MAPPING]
             raise GenerationError(
-                "unsupported reduction access '{0}' found in DynBuiltin:"
-                "reduction_sum_loop(). Expected one of '{1}'".
+                "Unsupported reduction access '{0}' found in LFRicBuiltIn:"
```

Comment:
Please could you change this to use `six.raise_from()`

Reply:
Done.